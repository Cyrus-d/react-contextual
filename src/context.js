import React from 'react'
import { createContext } from 'react-broadcast'

const providers = new Map()
const Context = createContext()

export function createNamedContext(name, initialState) {
    const context = createContext(initialState)
    providers.set(name, context)
    return context
}

export function getNamedContext(name) {
    return providers.get(name)
}

export function removeNamedContext(name) {
    providers.delete(name)
}

export function resolveContext(context, props) {
    let result
    if (typeof context === 'function') {
        // Test against component-symbol first, then assume a user function
        result = getNamedContext(context) || resolveContext(context(props))
    } else if (typeof context === 'string') {
        result = getNamedContext(context)
    }
    return result || context ||Â Context
}

export default Context