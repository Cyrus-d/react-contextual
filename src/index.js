import { subscribe, Subscribe } from './subscribe'
import { RenderPure, Provider } from './store'
import Context, {
    namedContext,
    moduleContext,
    transformContext,
    createNamedContext,
    removeNamedContext,
    getNamedContext,
    resolveContext,
} from './context'

export {
    Context,
    subscribe,
    Subscribe,
    RenderPure,
    Provider,
    namedContext,
    moduleContext,
    transformContext,
    createNamedContext,
    removeNamedContext,
    getNamedContext,
}
