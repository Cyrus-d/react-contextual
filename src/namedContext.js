import React from 'react'
import {
  resolveContext,
  createNamedContext,
  removeNamedContext,
} from './context'

export default function namedContext(contextName, initialState) {
  return Wrapped =>
    class NamedContext extends React.PureComponent {
      constructor(props) {
        super()
        this.name =
          typeof contextName === 'function' ? contextName(props) : contextName
        this.state = { context: createNamedContext(this.name, initialState) }
      }
      componentWillUnmount() {
        removeNamedContext(this.name)
      }
      render() {
        return <Wrapped {...this.props} context={this.state.context} />
      }
    }
}
