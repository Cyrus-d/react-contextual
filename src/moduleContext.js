import React from 'react'
import { createNamedContext } from './context'

export default function moduleContext(initialState) {
    return Wrapped => {
        let context = undefined
        const ModuleContext = class extends React.PureComponent {
            render() {
                return <Wrapped {...this.props} context={context} />
            }
        }
        context = createNamedContext(ModuleContext, initialState)
        return ModuleContext
    }
}