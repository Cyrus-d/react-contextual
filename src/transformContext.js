import React from 'react'
import { subscribe } from './subscribe'
import { resolveContext } from './context'

export default function transformContext(context, transform) {
    return Wrapped => {
        return subscribe(context, transform)(class extends React.PureComponent {
            render() {
                const context = resolveContext(context, this.props)
                return <Wrapped {...this.props} context={context} />
            }
        })
    }
}