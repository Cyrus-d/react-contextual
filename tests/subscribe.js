import React from 'react'
import { Provider, createStore, ProviderContext, Subscribe, subscribe, moduleContext } from '../src/'

const store = createStore({ initialState: { message: 'success!' } }, 'testStore')

Object.entries({
    'subscribe()': {},
    'subscribe(store => store)': { select: store => store },
    'subscribe("state")': { select: 'state' },
    'subscribe(ProviderContext)': { to: ProviderContext },
    'subscribe(ProviderContext, context => context)': { to: ProviderContext, select: context => context },
    'subscribe(ProviderContext, "state")': { to: ProviderContext, select: 'state' },
    'subscribe("key")': { id: 'key', select: 'state' },
    'subscribe("key", store => store)': { id: 'key', select: store => store },
    'subscribe("key", "state")': { id: 'key', select: 'state' },
    'subscribe(store)': { store, to: store },
    'subscribe(store, context => context)': { store, to: store, select: context => context },
    'subscribe(store, "state")': { store, to: store, select: 'state' },
    'subscribe(props => props.id)': { id: 'key', to: props => props.id, select: 'state' },
    'subscribe(props => props.id, store => store)': { id: 'key', to: props => props.id, select: store => store },
    'subscribe(props => props.id, "state")': { id: 'key', to: props => props.id, select: 'state' },
}).forEach(([key, value]) =>
    test(key, async () => {
        let { id, to, select, ...rest } = value
        const store = { initialState: { message: 'success!' } }
        const Test = subscribe(to || id, select)(props => props.message || props.state.message)
        await snapshot(
            <Provider id={id} {...store} {...rest}>
                <Test id={id} />
                <Subscribe
                    id={id}
                    to={to || id}
                    select={select}
                    children={props => props.message || props.state.message}
                />
            </Provider>,
        )
    }),
)

Object.entries({
    'subscribe(AnyContextObjectValue)': {},
    'subscribe(AnyContextSingleValue)': { state: 'success' },
    'subscribe(AnyContext, context => context)': { select: context => context },
    'subscribe(AnyContext, "context")': { select: 'context' },
}).forEach(([key, value]) =>
    test(key, async () => {
        let { state, select } = value
        state = state || { context: 'success!' }
        const AnyContext = moduleContext()(props => <props.context.Provider value={state} children={props.children} />)
        const Test = subscribe(AnyContext, select)(props => props.context.context || props.context)
        await snapshot(
            <AnyContext>
                <Test />
                <Subscribe to={AnyContext} select={select} children={props => props.context.context || props.context} />
            </AnyContext>,
        )
    }),
)
