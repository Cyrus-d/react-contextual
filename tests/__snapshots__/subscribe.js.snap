// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe("key") 1`] = `
<Provider
  id="key"
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap
        id="key"
      >
        <Consumer
          quiet={false}
        >
          <Component
            id="key"
            state={
              Object {
                "actions": Object {
                  "setState": [Function],
                },
                "message": "success!",
              }
            }
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        id="key"
        select="state"
        to="key"
      >
        <SubscribeWrap
          id="key"
        >
          <Consumer
            quiet={false}
          >
            <Component
              id="key"
              state={
                Object {
                  "actions": Object {
                    "setState": [Function],
                  },
                  "message": "success!",
                }
              }
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe("key", "state") 1`] = `
<Provider
  id="key"
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap
        id="key"
      >
        <Consumer
          quiet={false}
        >
          <Component
            id="key"
            state={
              Object {
                "actions": Object {
                  "setState": [Function],
                },
                "message": "success!",
              }
            }
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        id="key"
        select="state"
        to="key"
      >
        <SubscribeWrap
          id="key"
        >
          <Consumer
            quiet={false}
          >
            <Component
              id="key"
              state={
                Object {
                  "actions": Object {
                    "setState": [Function],
                  },
                  "message": "success!",
                }
              }
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe("key", store => store) 1`] = `
<Provider
  id="key"
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap
        id="key"
      >
        <Consumer
          quiet={false}
        >
          <Component
            actions={
              Object {
                "setState": [Function],
              }
            }
            id="key"
            message="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        id="key"
        select={[Function]}
        to="key"
      >
        <SubscribeWrap
          id="key"
        >
          <Consumer
            quiet={false}
          >
            <Component
              actions={
                Object {
                  "setState": [Function],
                }
              }
              id="key"
              message="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe("state") 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            state={
              Object {
                "actions": Object {
                  "setState": [Function],
                },
                "message": "success!",
              }
            }
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select="state"
        to={
          Object {
            "Consumer": [Function],
            "Provider": [Function],
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              state={
                Object {
                  "actions": Object {
                    "setState": [Function],
                  },
                  "message": "success!",
                }
              }
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe() 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            actions={
              Object {
                "setState": [Function],
              }
            }
            message="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={
          Object {
            "Consumer": [Function],
            "Provider": [Function],
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              actions={
                Object {
                  "setState": [Function],
                }
              }
              message="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe([a,b], (a,b) => props) 1`] = `
<Provider
  actions={
    Object {
      "up": [Function],
    }
  }
  id="id1"
  initialState={
    Object {
      "count": 0,
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "up": [Function],
        },
        "count": 0,
      }
    }
  >
    <RenderPure>
      <Provider
        actions={
          Object {
            "up": [Function],
          }
        }
        id="id2"
        initialState={
          Object {
            "count": 0,
          }
        }
      >
        <Provider
          value={
            Object {
              "actions": Object {
                "up": [Function],
              },
              "count": 0,
            }
          }
        >
          <RenderPure>
            <SubscribeWrap>
              <Consumer
                quiet={false}
              >
                <Consumer
                  quiet={false}
                >
                  <Component
                    store1={
                      Object {
                        "actions": Object {
                          "up": [Function],
                        },
                        "count": 0,
                      }
                    }
                    store2={
                      Object {
                        "actions": Object {
                          "up": [Function],
                        },
                        "count": 0,
                      }
                    }
                  >
                    <button
                      onClick={[Function]}
                    >
                      0
                       
                      0
                    </button>
                  </Component>
                </Consumer>
              </Consumer>
            </SubscribeWrap>
            <Subscribe
              select={[Function]}
              to={
                Array [
                  "id1",
                  "id2",
                ]
              }
            >
              <SubscribeWrap>
                <Consumer
                  quiet={false}
                >
                  <Consumer
                    quiet={false}
                  >
                    <Component
                      store1={
                        Object {
                          "actions": Object {
                            "up": [Function],
                          },
                          "count": 0,
                        }
                      }
                      store2={
                        Object {
                          "actions": Object {
                            "up": [Function],
                          },
                          "count": 0,
                        }
                      }
                    >
                      0 0
                    </Component>
                  </Consumer>
                </Consumer>
              </SubscribeWrap>
            </Subscribe>
          </RenderPure>
        </Provider>
      </Provider>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe([a,b], (a,b) => props) 2`] = `
<Provider
  actions={
    Object {
      "up": [Function],
    }
  }
  id="id1"
  initialState={
    Object {
      "count": 0,
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "up": [Function],
        },
        "count": 1,
      }
    }
  >
    <RenderPure>
      <Provider
        actions={
          Object {
            "up": [Function],
          }
        }
        id="id2"
        initialState={
          Object {
            "count": 0,
          }
        }
      >
        <Provider
          value={
            Object {
              "actions": Object {
                "up": [Function],
              },
              "count": 1,
            }
          }
        >
          <RenderPure>
            <SubscribeWrap>
              <Consumer
                quiet={false}
              >
                <Consumer
                  quiet={false}
                >
                  <Component
                    store1={
                      Object {
                        "actions": Object {
                          "up": [Function],
                        },
                        "count": 1,
                      }
                    }
                    store2={
                      Object {
                        "actions": Object {
                          "up": [Function],
                        },
                        "count": 1,
                      }
                    }
                  >
                    <button
                      onClick={[Function]}
                    >
                      1
                       
                      1
                    </button>
                  </Component>
                </Consumer>
              </Consumer>
            </SubscribeWrap>
            <Subscribe
              select={[Function]}
              to={
                Array [
                  "id1",
                  "id2",
                ]
              }
            >
              <SubscribeWrap>
                <Consumer
                  quiet={false}
                >
                  <Consumer
                    quiet={false}
                  >
                    <Component
                      store1={
                        Object {
                          "actions": Object {
                            "up": [Function],
                          },
                          "count": 1,
                        }
                      }
                      store2={
                        Object {
                          "actions": Object {
                            "up": [Function],
                          },
                          "count": 1,
                        }
                      }
                    >
                      1 1
                    </Component>
                  </Consumer>
                </Consumer>
              </SubscribeWrap>
            </Subscribe>
          </RenderPure>
        </Provider>
      </Provider>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe([a,b], [a,b]) 1`] = `
<Provider
  actions={
    Object {
      "up": [Function],
    }
  }
  id="id1"
  initialState={
    Object {
      "count": 0,
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "up": [Function],
        },
        "count": 0,
      }
    }
  >
    <RenderPure>
      <Provider
        actions={
          Object {
            "up": [Function],
          }
        }
        id="id2"
        initialState={
          Object {
            "count": 0,
          }
        }
      >
        <Provider
          value={
            Object {
              "actions": Object {
                "up": [Function],
              },
              "count": 0,
            }
          }
        >
          <RenderPure>
            <SubscribeWrap>
              <Consumer
                quiet={false}
              >
                <Consumer
                  quiet={false}
                >
                  <Component
                    store1={
                      Object {
                        "actions": Object {
                          "up": [Function],
                        },
                        "count": 0,
                      }
                    }
                    store2={
                      Object {
                        "actions": Object {
                          "up": [Function],
                        },
                        "count": 0,
                      }
                    }
                  >
                    <button
                      onClick={[Function]}
                    >
                      0
                       
                      0
                    </button>
                  </Component>
                </Consumer>
              </Consumer>
            </SubscribeWrap>
            <Subscribe
              select={
                Array [
                  "store1",
                  "store2",
                ]
              }
              to={
                Array [
                  "id1",
                  "id2",
                ]
              }
            >
              <SubscribeWrap>
                <Consumer
                  quiet={false}
                >
                  <Consumer
                    quiet={false}
                  >
                    <Component
                      store1={
                        Object {
                          "actions": Object {
                            "up": [Function],
                          },
                          "count": 0,
                        }
                      }
                      store2={
                        Object {
                          "actions": Object {
                            "up": [Function],
                          },
                          "count": 0,
                        }
                      }
                    >
                      0 0
                    </Component>
                  </Consumer>
                </Consumer>
              </SubscribeWrap>
            </Subscribe>
          </RenderPure>
        </Provider>
      </Provider>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe([a,b], [a,b]) 2`] = `
<Provider
  actions={
    Object {
      "up": [Function],
    }
  }
  id="id1"
  initialState={
    Object {
      "count": 0,
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "up": [Function],
        },
        "count": 1,
      }
    }
  >
    <RenderPure>
      <Provider
        actions={
          Object {
            "up": [Function],
          }
        }
        id="id2"
        initialState={
          Object {
            "count": 0,
          }
        }
      >
        <Provider
          value={
            Object {
              "actions": Object {
                "up": [Function],
              },
              "count": 1,
            }
          }
        >
          <RenderPure>
            <SubscribeWrap>
              <Consumer
                quiet={false}
              >
                <Consumer
                  quiet={false}
                >
                  <Component
                    store1={
                      Object {
                        "actions": Object {
                          "up": [Function],
                        },
                        "count": 1,
                      }
                    }
                    store2={
                      Object {
                        "actions": Object {
                          "up": [Function],
                        },
                        "count": 1,
                      }
                    }
                  >
                    <button
                      onClick={[Function]}
                    >
                      1
                       
                      1
                    </button>
                  </Component>
                </Consumer>
              </Consumer>
            </SubscribeWrap>
            <Subscribe
              select={
                Array [
                  "store1",
                  "store2",
                ]
              }
              to={
                Array [
                  "id1",
                  "id2",
                ]
              }
            >
              <SubscribeWrap>
                <Consumer
                  quiet={false}
                >
                  <Consumer
                    quiet={false}
                  >
                    <Component
                      store1={
                        Object {
                          "actions": Object {
                            "up": [Function],
                          },
                          "count": 1,
                        }
                      }
                      store2={
                        Object {
                          "actions": Object {
                            "up": [Function],
                          },
                          "count": 1,
                        }
                      }
                    >
                      1 1
                    </Component>
                  </Consumer>
                </Consumer>
              </SubscribeWrap>
            </Subscribe>
          </RenderPure>
        </Provider>
      </Provider>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(AnyContext, "context") 1`] = `
<ModuleContext>
  <Component
    context={
      Object {
        "Consumer": [Function],
        "Provider": [Function],
      }
    }
  >
    <Provider
      value={
        Object {
          "context": "success!",
        }
      }
    >
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            context={
              Object {
                "context": "success!",
              }
            }
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select="context"
        to={[Function]}
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              context={
                Object {
                  "context": "success!",
                }
              }
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </Provider>
  </Component>
</ModuleContext>
`;

exports[`subscribe(AnyContext, context => context) 1`] = `
<ModuleContext>
  <Component
    context={
      Object {
        "Consumer": [Function],
        "Provider": [Function],
      }
    }
  >
    <Provider
      value={
        Object {
          "context": "success!",
        }
      }
    >
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            context="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={[Function]}
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              context="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </Provider>
  </Component>
</ModuleContext>
`;

exports[`subscribe(AnyContextObjectValue) 1`] = `
<ModuleContext>
  <Component
    context={
      Object {
        "Consumer": [Function],
        "Provider": [Function],
      }
    }
  >
    <Provider
      value={
        Object {
          "context": "success!",
        }
      }
    >
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            context="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={[Function]}
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              context="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </Provider>
  </Component>
</ModuleContext>
`;

exports[`subscribe(AnyContextSingleValue) 1`] = `
<ModuleContext>
  <Component
    context={
      Object {
        "Consumer": [Function],
        "Provider": [Function],
      }
    }
  >
    <Provider
      value="success"
    >
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            context="success"
          >
            success
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={[Function]}
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              context="success"
            >
              success
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </Provider>
  </Component>
</ModuleContext>
`;

exports[`subscribe(ProviderContext) 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            actions={
              Object {
                "setState": [Function],
              }
            }
            message="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={
          Object {
            "Consumer": [Function],
            "Provider": [Function],
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              actions={
                Object {
                  "setState": [Function],
                }
              }
              message="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(ProviderContext, "state") 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            state={
              Object {
                "actions": Object {
                  "setState": [Function],
                },
                "message": "success!",
              }
            }
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select="state"
        to={
          Object {
            "Consumer": [Function],
            "Provider": [Function],
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              state={
                Object {
                  "actions": Object {
                    "setState": [Function],
                  },
                  "message": "success!",
                }
              }
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(ProviderContext, context => context) 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            actions={
              Object {
                "setState": [Function],
              }
            }
            message="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={
          Object {
            "Consumer": [Function],
            "Provider": [Function],
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              actions={
                Object {
                  "setState": [Function],
                }
              }
              message="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(props => props.id) 1`] = `
<Provider
  id="key"
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap
        id="key"
      >
        <Consumer
          quiet={false}
        >
          <Component
            id="key"
            state={
              Object {
                "actions": Object {
                  "setState": [Function],
                },
                "message": "success!",
              }
            }
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        id="key"
        select="state"
        to={[Function]}
      >
        <SubscribeWrap
          id="key"
        >
          <Consumer
            quiet={false}
          >
            <Component
              id="key"
              state={
                Object {
                  "actions": Object {
                    "setState": [Function],
                  },
                  "message": "success!",
                }
              }
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(props => props.id, "state") 1`] = `
<Provider
  id="key"
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap
        id="key"
      >
        <Consumer
          quiet={false}
        >
          <Component
            id="key"
            state={
              Object {
                "actions": Object {
                  "setState": [Function],
                },
                "message": "success!",
              }
            }
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        id="key"
        select="state"
        to={[Function]}
      >
        <SubscribeWrap
          id="key"
        >
          <Consumer
            quiet={false}
          >
            <Component
              id="key"
              state={
                Object {
                  "actions": Object {
                    "setState": [Function],
                  },
                  "message": "success!",
                }
              }
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(props => props.id, store => store) 1`] = `
<Provider
  id="key"
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap
        id="key"
      >
        <Consumer
          quiet={false}
        >
          <Component
            actions={
              Object {
                "setState": [Function],
              }
            }
            id="key"
            message="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        id="key"
        select={[Function]}
        to={[Function]}
      >
        <SubscribeWrap
          id="key"
        >
          <Consumer
            quiet={false}
          >
            <Component
              actions={
                Object {
                  "setState": [Function],
                }
              }
              id="key"
              message="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(store => store) 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            actions={
              Object {
                "setState": [Function],
              }
            }
            message="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={
          Object {
            "Consumer": [Function],
            "Provider": [Function],
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              actions={
                Object {
                  "setState": [Function],
                }
              }
              message="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(store) 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
  store={
    Object {
      "actions": Object {
        "setState": [Function],
      },
      "context": Object {
        "Consumer": [Function],
        "Provider": [Function],
      },
      "destroy": [Function],
      "getState": [Function],
      "id": "testStore",
      "initialState": Object {
        "message": "success!",
      },
      "props": Object {},
      "state": Object {
        "message": "success!",
      },
      "subscribe": [Function],
      "subscriptions": Set {},
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            actions={
              Object {
                "setState": [Function],
              }
            }
            message="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={
          Object {
            "actions": Object {
              "setState": [Function],
            },
            "context": Object {
              "Consumer": [Function],
              "Provider": [Function],
            },
            "destroy": [Function],
            "getState": [Function],
            "id": "testStore",
            "initialState": Object {
              "message": "success!",
            },
            "props": Object {},
            "state": Object {
              "message": "success!",
            },
            "subscribe": [Function],
            "subscriptions": Set {},
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              actions={
                Object {
                  "setState": [Function],
                }
              }
              message="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(store, "state") 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
  store={
    Object {
      "actions": Object {
        "setState": [Function],
      },
      "context": Object {
        "Consumer": [Function],
        "Provider": [Function],
      },
      "destroy": [Function],
      "getState": [Function],
      "id": "testStore",
      "initialState": Object {
        "message": "success!",
      },
      "props": Object {},
      "state": Object {
        "message": "success!",
      },
      "subscribe": [Function],
      "subscriptions": Set {},
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            state={
              Object {
                "actions": Object {
                  "setState": [Function],
                },
                "message": "success!",
              }
            }
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select="state"
        to={
          Object {
            "actions": Object {
              "setState": [Function],
            },
            "context": Object {
              "Consumer": [Function],
              "Provider": [Function],
            },
            "destroy": [Function],
            "getState": [Function],
            "id": "testStore",
            "initialState": Object {
              "message": "success!",
            },
            "props": Object {},
            "state": Object {
              "message": "success!",
            },
            "subscribe": [Function],
            "subscriptions": Set {},
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              state={
                Object {
                  "actions": Object {
                    "setState": [Function],
                  },
                  "message": "success!",
                }
              }
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;

exports[`subscribe(store, context => context) 1`] = `
<Provider
  initialState={
    Object {
      "message": "success!",
    }
  }
  store={
    Object {
      "actions": Object {
        "setState": [Function],
      },
      "context": Object {
        "Consumer": [Function],
        "Provider": [Function],
      },
      "destroy": [Function],
      "getState": [Function],
      "id": "testStore",
      "initialState": Object {
        "message": "success!",
      },
      "props": Object {},
      "state": Object {
        "message": "success!",
      },
      "subscribe": [Function],
      "subscriptions": Set {},
    }
  }
>
  <Provider
    value={
      Object {
        "actions": Object {
          "setState": [Function],
        },
        "message": "success!",
      }
    }
  >
    <RenderPure>
      <SubscribeWrap>
        <Consumer
          quiet={false}
        >
          <Component
            actions={
              Object {
                "setState": [Function],
              }
            }
            message="success!"
          >
            success!
          </Component>
        </Consumer>
      </SubscribeWrap>
      <Subscribe
        select={[Function]}
        to={
          Object {
            "actions": Object {
              "setState": [Function],
            },
            "context": Object {
              "Consumer": [Function],
              "Provider": [Function],
            },
            "destroy": [Function],
            "getState": [Function],
            "id": "testStore",
            "initialState": Object {
              "message": "success!",
            },
            "props": Object {},
            "state": Object {
              "message": "success!",
            },
            "subscribe": [Function],
            "subscriptions": Set {},
          }
        }
      >
        <SubscribeWrap>
          <Consumer
            quiet={false}
          >
            <Component
              actions={
                Object {
                  "setState": [Function],
                }
              }
              message="success!"
            >
              success!
            </Component>
          </Consumer>
        </SubscribeWrap>
      </Subscribe>
    </RenderPure>
  </Provider>
</Provider>
`;
